src/utils/Api.js

```diff
@@ -0,0 +1,136 @@
+class Api {
+  constructor({ baseUrl, authToken }) {
+    this._baseUrl = baseUrl;
+    this._authToken = authToken;
+  }
+
+  getUsers = () =>
+    fetch(`${this._baseUrl}/users/me`, {
+      method: "GET",
+      headers: {
+        authorization: this._authToken,
+        "Content-Type": "application/json",
+      },
+    }).then((res) => {
+      if (res.ok) {
+        return res.json();
+      }
+      // if the server returns an error, reject the promise
+      return Promise.reject(`Error: ${res.status}`);
+    });
+
+  editUserInfo = (name, about) =>
+    fetch(`${this._baseUrl}/users/me`, {
+      method: "PATCH",
+      headers: {
+        "Content-Type": "application/json", // Include Content-Type header
+        authorization: this._authToken,
+      },
+      body: JSON.stringify({
+        name: name, // Use the parameters directly
+        about: about,
+      }),
+    })
+      .then((res) => {
+        if (res.ok) {
+          return res.json();
+        }
+        // If the server returns an error, reject the promise with a status message
+        return Promise.reject(new Error(`Error: ${res.status}`));
+      })
+      .catch((error) => console.error("Request failed", error)); // Catch and log any errors
```

Comment:
You should **not** use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request, and this end is in `index.js`   after all `then` blocks. And in real projects the `catch` blocks usually open error popups and so on.